content/ember-v4-typescript-support-update.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Ember v4 TypeScript Support Update
+authors:
+  - chriskrycho
+date: 2022-01-25T00:00:00.000Z
+tags:
+  - '2022'
+  - typescript
+
+---
+
+The [Typed Ember](https://github.com/typed-ember) team—[Chris Krycho (@chriskrycho)](https://github.com/chriskrycho), [Dan Freeman (@dfreeman)](https://github.com/dfreeman), and [James Davis (@jamescdavis)](https://github.com/jamescdavis)—is happy to announce that the `@types` type definition packages, maintained on [Definitely Typed](https://github.com/DefinitelyTyped/DefinitelyTyped), now have full support for Ember 4.x! This is *two major changes* with *two big benefits*.
+
+The two major changes:
+
+* **All APIs which were deprecated during the 3.x cycle and removed in Ember v4.0.0 have been dropped from the types.** (Amusingly, this includes a number of things which were deprecated in 2.x and removed at 3.0, and even a few which were deprecated the 1.x era and removed at 2.0!) Additionally, a number of types representing private API have been removed, in line with our standing policy of providing types for *only* public APIs.
+
+* **The minimum supported TypeScript version is 4.4.** This aligns with the direction we expect to propose for Ember’s own TypeScript support policy, where major releases will generally adopt a recent TypeScript version as their minimum supported version, and with a support policy similar to that for browsers and Node LTS releases. (Keep your eyes open for a forthcoming RFC tackling those details!)
+
+[e-typescript]: https://discord.com/channels/480462759797063690/484421406659182603
+
+The two big benefits:
+
+1. **If you’re a JavaScript Ember user**, your editor tooling may get a bit smarter and better invisibly: VS Code (and maybe other editors, too) can take advantage of these types to provide you with better autocomplete, docs, etc. and those will all correctly reflect Ember v4 APIs.
+
+2. **If you’re a TypeScript Ember user**, you can now use v4 in your `@types` dependencies. That will give you the same benefits as JavaScript users get via the up-to-date definitions for Ember itself, but it will also provide *much* better feedback and errors in many cases, because we are now using the latest and greatest features of TypeScript for these types.
```

Comment:
"but it will also provide much better feedback and errors in many cases" reads a bit strange for me. What's a better error? I don't want to get errors 😂!

Reply:
```suggestion
2. **If you’re a TypeScript Ember user**, you can now use v4 in your `@types` dependencies. That will give you the same benefits as JavaScript users get via the up-to-date definitions for Ember itself, but it will also catch more errors and provide *much* better feedback in many cases, because we are now using the latest and greatest features of TypeScript for these types.
```